compiler/modules/bir/bir.bal

```diff
@@ -176,8 +176,37 @@ public type RegisterBase record {|
 public type DeclRegister ParamRegister|VarRegister|FinalRegister;
 public type Register DeclRegister|NarrowRegister|TmpRegister|AssignTmpRegister;
 
+public const LEXICAL_BLOCK_SCOPE_KIND = "lexical_scope";
+public const FUNCTION_SCOPE_KIND = "function_scope";
+public type RegisterScopeKind LEXICAL_BLOCK_SCOPE_KIND|FUNCTION_SCOPE_KIND;
+
+# Represent scope inside the function body such as a while loop
+public type LexicalBlockScope readonly & record {|
+    LEXICAL_BLOCK_SCOPE_KIND kind = LEXICAL_BLOCK_SCOPE_KIND;
+    RegisterScope parentScope;
+    Position startPos;
+    Position endPos;
+|};
+
+# We don't need a reference to the actual function here since the scaffold will have it.
+public type FunctionScope readonly & record {|
```

Comment:
Why do we need this?  Can't we use just use `LexicalBlockScope?` for scope?

Reply:
Fixed (Removed)